Update doc.mk for recent xsltproc

In the 10 years since this code was last touched, xsltproc/docbook has
made some changes.
1. http://docbook.sourceforge.net/release/xsl-ns/current/html/docbook.xsl
   must have fallen from favor, it is no longer listed in 
   /etc/xml/catalog, instead there is
   http://docbook.sourceforge.net/release/xsl/current/html/docbook.xsl
2. '-' does not work as indicating stdin

This is a terrible hack, using '<target>.tmp' file.

diff --git a/doc/doc.mk b/doc/doc.mk
index bde1e0e..6fbdab4 100644
--- a/doc/doc.mk
+++ b/doc/doc.mk
@@ -29,10 +29,11 @@ EDIT_XML = $(XSLTPROC) \
 if BUILD_DOCS
 
 .xml.html:
-	$(V_DOCBOOK)$(EDIT_XML) $< | $(XSLTPROC) $(DOCBOOK_PARAMS) \
+	$(V_DOCBOOK)$(EDIT_XML) $< > $@.tmp
+	$(XSLTPROC) $(DOCBOOK_PARAMS) \
   --stringparam html.stylesheet html.css \
-  http://docbook.sourceforge.net/release/xsl-ns/current/html/docbook.xsl \
-  - > $@
+  http://docbook.sourceforge.net/release/xsl/current/html/docbook.xsl \
+  $@.tmp > $@
 
 .xml.fo:
 	$(V_DOCBOOK)$(EDIT_XML) $< | $(XSLTPROC) $(DOCBOOK_PARAMS) $(srcdir)/doc/xmlstar-fodoc-style.xsl - > $@
@@ -49,9 +50,10 @@ endif
 $(userguide).html : $(userguide_src)
 
 $(manpage): $(manpage_src)
-	 $(V_DOCBOOK)$(EDIT_XML) $< | $(XSLTPROC) -o $@ \
-	  http://docbook.sourceforge.net/release/xsl-ns/current/manpages/docbook.xsl \
-	  -
+	 $(V_DOCBOOK)$(EDIT_XML) $< > $@.tmp
+	 $(XSLTPROC) -o $@ \
+	  http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl \
+	  $@.tmp
 
 $(txtguide): $(txtguide_src) $(usage_texts)
 	$(AM_V_GEN)srcdir=$(srcdir) SED=$(SED) transform='$(program_transform_name)' $< ./xml > $@
diff --git a/doc/xmlstar-fodoc-style.xsl b/doc/xmlstar-fodoc-style.xsl
index 11eaedb..cd8f8e7 100644
--- a/doc/xmlstar-fodoc-style.xsl
+++ b/doc/xmlstar-fodoc-style.xsl
@@ -1,7 +1,7 @@
 <?xml version='1.0'?>
 <xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="1.0">
 
-  <xsl:import href="http://docbook.sourceforge.net/release/xsl-ns/current/fo/docbook.xsl"/>
+  <xsl:import href="http://docbook.sourceforge.net/release/xsl/current/fo/docbook.xsl"/>
 
   <xsl:param name="fop1.extensions" select="1"/>
 
